### PR TITLE
feat: added support for call value

### LINKED ISSUE #114: feat request: add reading calls msg data and value
It'd be great to know when a mock is called, what was the raw shape of the `msg` structure.
In our case, we need to check that a `payable` function was called with the correct amount of funds.

### PR DESCRIPTION
**Pull request by 0xGorilla**
**Description**
Adds full support for call `msg.value`, see changed documentation in order to see new implemented methods

**Metadata**

- Fixes #114

### CODE REVIEW

File: src/factories/smock-contract.ts [hunk 1]
```diff
@@ -205,6 +205,7 @@ function parseMessage(message: Message, contractInterface: Interface, sighash: s
   return {
     args: sighash === null ? toHexString(message.data) : getMessageArgs(message.data, contractInterface, sighash),
     nonce: Sandbox.getNextNonce(),
+    value: BigNumber.from(message.value.toString()),
```
Comment (wei3erHase) [inline src/factories/smock-contract.ts:208]: what's the type of `message.value`?

Comment (0xGorilla) [reply src/factories/smock-contract.ts:208]: BN using from library `bn.js`

File: test/unit/watchable-function-logic/type-handling.spec.ts [hunk 2]
```diff
@@ -123,4 +127,15 @@ describe('WatchableFunctionLogic: Type handling', () => {
     fake['receiveOverload(bool)'].should.have.been.calledWith(true);
     fake['receiveOverload(bool,bool)'].should.have.been.calledWith(true, false);
   });
+
+  it('should handle msg.value', async () => {
+    const value = BigNumber.from(123);
+    await fake.connect(signer).receiveEmpty({ value });
+    expect(fake.receiveEmpty.getCall(0).value).to.eq(value);
```
Comment (wei3erHase) [inline test/unit/watchable-function-logic/type-handling.spec.ts:134]: why are we not using expect before?

**Review by wei3erHase — APPROVED** [review]
looks great pal!
made some questions but ready to go 🚀

File: src/factories/smock-contract.ts (hunk 1 above)
Comment (wei3erHase) [reply src/factories/smock-contract.ts:208]: puajj

File: test/unit/watchable-function-logic/type-handling.spec.ts (hunk 2 above)
Comment (0xGorilla) [reply test/unit/watchable-function-logic/type-handling.spec.ts]: using should now

**Pull request merged**